src/onelogin/saml2/settings.py

```diff
@@ -557,18 +507,14 @@ def get_sp_cert(self):
         :returns: SP public cert
         :rtype: string
         """
-        cert = None
+        cert = self.__sp.get('x509cert')
+        cert_file_name = self.__paths['cert'] + 'sp.crt'
 
-        if 'x509cert' in self.__sp.keys() and self.__sp['x509cert']:
-            cert = self.__sp['x509cert']
-        else:
-            cert_file_name = self.__paths['cert'] + 'sp.crt'
-            if exists(cert_file_name):
-                f_cert = open(cert_file_name, 'r')
-                cert = f_cert.read()
-                f_cert.close()
+        if not cert and exists(cert_file_name):
+            with open(cert_file_name) as f:
+                cert = f.read()
 
-        return cert
+        return cert or None  # TODO: Should change rtype docstrings or update tests
```

Comment:
I don't think is a good idea, let's keep return types (and only change that if strictly necessary)